File: algorithm/array_and_matrices/271.py
from typing import List

class Solution:
    def encode(self, strs: List[str]) -> str:
        result = ""
        for s in strs:
            if len(s) < 10:
                result += "00" + str(len(s)) + s
            elif len(s) < 100:
                result += "0" + str(len(s)) + s
            else:
                result += str(len(s)) + s
        return result
    def decode(self, s: str) -> List[str]:
        result = []
        ptr = 0
        while ptr < len(s):
            num = int(s[ptr:ptr+3])
            result.append(s[ptr+3:ptr+3+num])
            ptr += 3 + num
        return result

        
if __name__ == "__main__":
    s = ["neet","code","love","you"]
    print(Solution().decode(Solution().encode(s)))

File: algorithm/array_and_matrices/271.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution
{
public:
    string encode(vector<string> &strs)
    {
        string result = "";
        for (const auto &s : strs)
        {
            if (s.size() < 10)
            {
                result += "00" + to_string(s.size()) + s;
            }
            else if (s.size() < 100)
            {
                result += "0" + to_string(s.size()) + s;
            }
            else
            {
                result += to_string(s.size()) + s;
            }
        }
        return result;
    }

    vector<string> decode(string s)
    {
        vector<string> result;
        int ptr = 0;
        int num = 0;
        while (ptr < s.size())
        {
            string num = s.substr(ptr, 3);
            result.push_back(s.substr(ptr + 3, stoi(num)));
            ptr += 3 + stoi(num);
        }
        return result;
    }
};
